addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1096,17 +1096,24 @@ const Wysiwyg = Widget.extend({
                             backgroundGradient = backgroundImage;
                         }
                     }
+                    const hadSelection = range && (range.startContainer !== range.endContainer || range.startOffset !== range.endOffset);
```

Comment:
You could probably have used `selection && selection.rangeCount && !selection.isCollapsed` (or `range && !selection.isCollapsed` in this case) instead which is more explicit.

Also here you're checking if it had a _non-collapsed_ selection, not if it had _a selection_ so the name is a little bit ambiguous.